include/pisa/wand_data_compressed.hpp

```diff
@@ -111,12 +112,12 @@ namespace {
                 spdlog::info("Storing max weight for each list and for each block...");
             }
 
-            float add_sequence(binary_freq_collection::sequence const &seq, binary_freq_collection const &coll, std::vector<float> const & norm_lens, const float lambda = 0.0f){
+            float add_sequence(binary_freq_collection::sequence const &seq, binary_freq_collection const &coll, std::vector<float> const & norm_lens, boost::variant<float, uint64_t> block_size) {
 
                 if (seq.docs.size() > configuration::get().threshold_wand_list) {
 
-                    auto t = ((type == partition_type::fixed_blocks) ? static_block_partition(seq, norm_lens)
-                                                      : variable_block_partition(coll, seq, norm_lens, lambda));
+                    auto t = ((type == partition_type::fixed_blocks) ? static_block_partition(seq, norm_lens, boost::get<uint64_t>(block_size))
```

Comment:
So typically, you would use a static visitor: https://www.boost.org/doc/libs/1_70_0/doc/html/variant/reference.html#variant.concepts.static-visitor

However, it is way too verbose for this, at least for inline use. Might as well do an `if` statement. If `type` not available, you can use `which` method that returns the index of the type that's in the variant. Not a great design though.

Not that for `std::variant`, it is very easy to write a wrapper function that turns this use case into:
```cpp
auto t = match(block_size,
    [](float alpha) { ... },
    [](uint64_t size) { ... })
```

You can see the example of that in on of the old commits in `warcpp`: https://github.com/pisa-engine/warcpp/blob/e8c53c09cb983ff6166360d46c514d8e36acb91a/include/warcpp/warcpp.hpp#L94 This **doesn't work** for our currently used AppleClang, though (blame Antonio).

It might be also possible for `boost::variant` but not sure how easy it is (the interface is slightly different). There is a lib for that: https://github.com/exclipy/inline_variant_visitor but it seems much longer compared to the `std::variant` version. And clearly, bringing another library for this is an overkill.

Reply:
I agree with this one!

Reply:
Yes, good point. I have used a similar idea that you mentioned below since we are using `boost` for now. Let's revisit this in #188 when we get to it.